### PR TITLE
Filtering Service

### LINKED ISSUE #24: Create Filtering Service
Acceptance Criteria :

Create the analytical backend to run queries for filtering.
Includes fast queries that post on the database.
Returns result in the valid format that can be used by the VizRec system.
Technical Notes :

use cube.dev
use mongodb as the underlying database.

### PR DESCRIPTION
**Pull request by ghostaccount737**
Closes #24

### CODE REVIEW

**Comment by lastlegion** [comment]
Please remove node_modules from git and add it to gitignore.

**Review by SapoonDutta — COMMENTED** [review]
Other than the binary files, the rest looks good to me.

File: query/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+querydb:
```
Comment (lastlegion) [inline query/docker-compose.yml]: why have a seperate docker-compose here?

File: ingestion/docker-compose.yml [hunk 2]
```diff
@@ -1,12 +1,12 @@
 web:
     build: .
-    command: python -u ingestion/run.py
+    command: python -u run.py
```
Comment (lastlegion) [inline ingestion/docker-compose.yml]: remove `ingestion/docker-compose.yml`
create `/docker-compose.yml` that brings up all the services

File: ingestion/docker-compose.yml [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
     ports:
         - "5000:5000"
     environment:
       DB_PORT_27017_TCP_ADDR: 27017
     volumes:
-        - .:/ingestion/app
+        - .:/app
```
Comment (lastlegion) [inline ingestion/docker-compose.yml]: not needed

File: ingestion/docker-compose.yml [hunk 4]
```diff
@@ -1,12 +1,12 @@
 web:
     build: .
-    command: python -u ingestion/run.py
+    command: python -u run.py
     ports:
         - "5000:5000"
     environment:
       DB_PORT_27017_TCP_ADDR: 27017
```
Comment (lastlegion) [inline ingestion/docker-compose.yml]: please rename to `DB_PORT`

File: docker-compose.yml [hunk 5]
```diff
@@ -1,15 +1,19 @@
-web:
-    build: .
+version: '2'
+services:
+  web:
+    build:
```
Comment (lastlegion) [inline docker-compose.yml:4]: bring up `query` aka filtering service as well.

File: docker-compose.yml [hunk 6]
```diff
@@ -1,15 +1,19 @@
...
+      dockerfile: ./ingestion/Dockerfile
     command: python -u ingestion/run.py
     ports:
         - "5000:5000"
     environment:
-      DB_PORT_27017_TCP_ADDR: 27017
+      DB_PORT: 27017
     volumes:
-        - .:/ingestion/app
+        - .:/app
```
Comment (lastlegion) [inline docker-compose.yml]: remove

**Review by lastlegion — APPROVED** [review]

**Pull request merged**